src/ast/expression.rs

```diff
@@ -3,25 +3,31 @@ use crate::{
   token::{Grammar, Keyword, Operator, TokenStream},
 };
 
-use super::{identifier::Identifier, AstError, AstResult, Location};
+use super::{
+  identifier::{self, Identifier},
+  AstError, AstResult, Location,
+};
 
 #[derive(Debug, PartialEq, Clone)]
 pub enum Expression<'a> {
   Bool(bool, Location),
   String(&'a str, Location), // TODO: given we want to reverse, maybe use owned?
   Number(f64, Location),
+  List(Vec<Expression<'a>>, Location),
+  Map(Vec<(Identifier, Expression<'a>)>, Location),
```

Comment:
`Identifier`s more represent a syntax variable name token, if maps are indexed with strings using `String` probably makes more sense.

`HashMap` might also be more appropriate than `Vec` here.